test/e2e/driver/azuredisk_driver.go

```diff
@@ -59,10 +59,22 @@ func (d *azureDiskDriver) GetDynamicProvisionStorageClass(parameters map[string]
 	provisioner := d.driverName
 	generateName := fmt.Sprintf("%s-%s-dynamic-sc-", namespace, normalizeProvisioner(provisioner))
 	var allowedTopologies []v1.TopologySelectorTerm
-	if skuName, ok := parameters[consts.SkuNameField]; ok && strings.HasSuffix(skuName, "_ZRS") {
-		mode := storagev1.VolumeBindingImmediate
-		bindingMode = &mode
-	} else {
+	isZRS := false
+	for key, value := range parameters {
+		if strings.EqualFold(key, consts.SkuNameField) {
+			if strings.HasSuffix(value, "_ZRS") {
+				mode := storagev1.VolumeBindingImmediate
+				bindingMode = &mode
+				isZRS = true
+			}
+
+			if strings.EqualFold(os.Getenv("AZURE_CLOUD_NAME"), "AZURESTACKCLOUD") && !strings.EqualFold(value, "Standard_LRS") && !strings.EqualFold(value, "Premium_LRS") {
```

Comment:
This can be `else if` because we know the value is not "Standard_LRS" or "Premium_LRS" if the first `if` clause was true.